go/ct/evm_fuzz_test.go

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package ct_test
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"pgregory.net/rand"
+
+	"github.com/Fantom-foundation/Tosca/go/ct"
+	. "github.com/Fantom-foundation/Tosca/go/ct/common"
+	"github.com/Fantom-foundation/Tosca/go/ct/st"
+	"github.com/Fantom-foundation/Tosca/go/vm"
+	"github.com/Fantom-foundation/Tosca/go/vm/geth"
+	"github.com/Fantom-foundation/Tosca/go/vm/lfvm"
+)
+
+// FuzzGeth is a fuzzing test for the geth EVM implementation
+// TODO: it would be interesting to have a method to extend the corpus of the fuzzer
+// with the failures found in other tests.
+// So far failures will be stored in the folder: testdata/fuzz/FuzzerFunctionName/
+func FuzzGeth(f *testing.F) {
+	fuzzVm(geth.NewConformanceTestingTarget(), f)
+}
+
+// FuzzLfvm is a fuzzing test for lfvm
+func FuzzLfvm(f *testing.F) {
+	fuzzVm(lfvm.NewConformanceTestingTarget(), f)
+}
+
+// FuzzLfvm is a fuzzing test for evmzero
+// func FuzzEvmzero(f *testing.F) {
+// 	fuzzVm(evmzero.NewConformanceTestingTarget(), f)
+// }
+
+// FuzzDifferentialLfvmVsGeth compares state output between lfvm and geth
+func FuzzDifferentialLfvmVsGeth(f *testing.F) {
+	differentialFuzz(f,
+		lfvm.NewConformanceTestingTarget(),
+		geth.NewConformanceTestingTarget(),
+	)
+}
+
+// TODO: This test makes sense but cannot be enabled yet:
+// - The evmzero fails the differential test against geth. (issue #549)
+// - Any other invocation of fuzzing tests in this file seem to
+// invoke the all other tests in the file, and they will fail.
+// func FuzzDifferentialEvmzeroVsGeth(f *testing.F) {
+// 	differentialFuzz(f,
+// 		evmzero.NewConformanceTestingTarget(),
+// 		geth.NewConformanceTestingTarget(),
+// 	)
+// }
+
+//////////////////////////////////////////////////////////////////////////////
+// Fuzzing helpers
+
+func differentialFuzz(f *testing.F, testeeVm ct.Evm, referenceVm ct.Evm) {
+
+	rnd := rand.New(0)
+
+	prepareFuzzingSeeds(f, rnd)
+
+	f.Fuzz(func(t *testing.T, opCodes []byte, gas int64, revision byte, stackBytes []byte) {
```

Comment:
why not use corresponding `vm`  package types for opcodes, gas, revision ?

Reply:
this is a requirement form the fuzzer, there is only a list of allowed types: 
-     string, []byte
-     int, int8, int16, int32/rune, int64
-     uint, uint8/byte, uint16, uint32, uint64
-     float32, float64
-     bool